plugin/trino-opa/src/test/java/io/trino/plugin/opa/OpaAccessControlUnitTest.java

```diff
@@ -0,0 +1,1459 @@
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Streams;
+import io.trino.spi.connector.CatalogSchemaName;
+import io.trino.spi.connector.CatalogSchemaRoutineName;
+import io.trino.spi.connector.CatalogSchemaTableName;
+import io.trino.spi.security.Identity;
+import io.trino.spi.security.PrincipalType;
+import io.trino.spi.security.Privilege;
+import io.trino.spi.security.SystemSecurityContext;
+import io.trino.spi.security.TrinoPrincipal;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Named;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.BiConsumer;
+import java.util.stream.Stream;
+
+import static com.google.common.collect.ImmutableSet.toImmutableSet;
+import static com.google.common.net.MediaType.JSON_UTF_8;
+import static io.trino.plugin.opa.HttpClientUtils.InstrumentedHttpClient;
+import static io.trino.plugin.opa.RequestTestUtilities.assertJsonRequestsEqual;
+import static io.trino.plugin.opa.RequestTestUtilities.assertStringRequestsEqual;
+import static io.trino.plugin.opa.TestHelpers.NO_ACCESS_RESPONSE;
+import static io.trino.plugin.opa.TestHelpers.OK_RESPONSE;
+import static io.trino.plugin.opa.TestHelpers.convertSystemSecurityContextToIdentityArgument;
+import static io.trino.plugin.opa.TestHelpers.createFailingTestCases;
+import static io.trino.plugin.opa.TestHelpers.createIllegalResponseTestCases;
+import static io.trino.plugin.opa.TestHelpers.systemSecurityContextFromIdentity;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class OpaAccessControlUnitTest
+{
+    private static final URI OPA_SERVER_URI = URI.create("http://my-uri/");
+    private InstrumentedHttpClient mockClient;
+    private OpaAccessControl authorizer;
+    private final JsonMapper jsonMapper = new JsonMapper();
+    private Identity requestingIdentity;
+    private SystemSecurityContext requestingSecurityContext;
+
+    @BeforeEach
+    public void setupAuthorizer()
+    {
+        this.mockClient = new InstrumentedHttpClient(OPA_SERVER_URI, "POST", JSON_UTF_8.toString(), request -> OK_RESPONSE);
+        this.authorizer = (OpaAccessControl) new OpaAccessControlFactory().create(ImmutableMap.of("opa.policy.uri", OPA_SERVER_URI.toString()), Optional.of(mockClient));
```

Comment:
This create method is a static factory method, so you don't need the `new`
```suggestion
        this.authorizer = (OpaAccessControl) OpaAccessControlFactory.create(ImmutableMap.of("opa.policy.uri", OPA_SERVER_URI.toString()), Optional.of(mockClient));
```